compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12845,195 +12845,302 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    }
 
 
+
+
+// Overflow/Underflow handling:
+// Must detect overflow without causing overflow:
+// Equation: add operation: x + Const1 < Const2 --> x < Const2 - Const1
+//          sub operation: x - Const1 < Const2 --> x < Const2 + Const1
+
+//Signed Constants may be negative, therefore underflow and overflow are possible for both add and sub operations
+bool isNotSignedOverflow(bool isAddOp, bool isSubOp, int64_t oldConst1, int64_t oldConst2, int64_t signedMax){
+   return (isAddOp && !(oldConst1 < 0 && (oldConst2 > signedMax + oldConst1)))
+            || (isSubOp && !(oldConst1 > 0 && (oldConst2 > signedMax - oldConst1)));
+}
+
+bool isNotSignedUnderflow(bool isAddOp, bool isSubOp, int64_t oldConst1, int64_t oldConst2, int64_t signedMin){
```

Comment:
can we pass in the opcode to test isAdd or something rather than two bools which is really ugly and hard to read?